src/core/sequence_batch_scheduler.cc

```diff
@@ -827,6 +843,82 @@ SequenceBatch::SetControlTensors(
   }
 }
 
+void
+SequenceBatch::UpdateImplicitState(
+    std::unique_ptr<InferenceRequest>& irequest, const int32_t seq_slot)
+{
+  // This should be executed only if the model has a states section.
+  if (base_->HasImplicitState()) {
+    std::shared_ptr<std::unordered_map<
+        std::string, std::unique_ptr<InferenceRequest::State>>>& states =
+        states_[seq_slot];
+
+    // Create the state for the first request in the sequence.
+    if (states == nullptr) {
+      states.reset(new std::unordered_map<
+                   std::string, std::unique_ptr<InferenceRequest::State>>{});
+      for (auto& state : base_->StateOutputConfigMap()) {
+        std::shared_ptr<InferenceRequest::Input> input;
+        auto& state_config = state.second;
+
+        std::vector<int64_t> dims;
+        for (auto& dim : state_config.dims()) {
+          if (dim == -1) {
+            dims.push_back(1);
+          } else {
+            dims.push_back(dim);
+          }
+        }
+
+        // Replace the dynamic shapes with 1 for the first inference request.
```

Comment:
This is so that the model can know beforehand what shape to expect for the first inference request for the implicit state. This was part of the proposal.

> Update comment explaining why this won't be an issue.

You mean add a comment explaining why the garbage data won't be an issue?

Reply:
You are right. I missed it. Make sure we document this somewhere.